odd-platform-api/src/test/java/org/opendatadiscovery/oddplatform/repository/ReactiveDataQualityRunsRepositoryTest.java

```diff
@@ -0,0 +1,214 @@
+    public void testGetMonitoredTables() {
+        final int numberOfNotMonitoredTables = 5;
+        IntStream.range(0, 5)
+                .forEach(i -> createHollowDataSentEntityTable());
+
+        final TablesDashboard tablesDashboard =
+                tablesDashboardMapper.mapToDto(List.of(), repository.getMonitoredTables()
+                        .collectList()
+                        .block());
+
+        assertEquals(numberOfNotMonitoredTables, tablesDashboard.getMonitoredTables().getNotMonitoredTables());
+        assertEquals(NUMBER_OF_ASSERTION + NUMBER_OF_SCHEMA_CHANGES
+                        + NUMBER_OF_COLUMN_ANOMALY + NUMBER_OF_FRESHNESS_ANOMALY + NUMBER_OF_VOLUME_ANOMALY,
+                tablesDashboard.getMonitoredTables().getMonitoredTables());
+    }
+
+    private void generateTestsData() {
+        ReactiveDataQualityRunsRepositoryTest.EXPECTED_VALUES
+                .forEach((key, value) -> generateSpecificTestData(key, value.getLeft(), value.getRight()));
+    }
+
+    private void generateSpecificTestData(final DataQualityCategory category,
+                                          final List<DataEntityRunStatus> statuses,
+                                          final int numberOfTests) {
+        IntStream.range(0, numberOfTests)
+                .mapToObj(value -> createHollowDataSentEntityTable())
+                .forEach(hollowEntity -> {
+                    statuses.forEach(status -> {
+                        final DataEntityPojo assertion = createDataQualityEntityWithCategory(category);
+                        dataQualityTestRelationRepository.createRelations(List.of(
+                                new DataQualityTestRelationsPojo()
+                                        .setDataQualityTestOddrn(assertion.getOddrn())
+                                        .setDatasetOddrn(hollowEntity.getOddrn())
+                        )).block();
+                        createTaskRun(assertion, status);
+                    });
+                });
+    }
+
+    private DataEntityPojo createHollowDataSentEntityTable() {
+        return dataEntityRepository
+                .create(new DataEntityPojo().setOddrn(UUID.randomUUID().toString())
+                        .setTypeId(DataEntityTypeDto.TABLE.getId()))
+                .block();
+    }
+
+    private DataEntityPojo createDataQualityEntityWithCategory(final DataQualityCategory category) {
+        final JSONB jsonb = JSONB.jsonb(
+                DATA_QUALITY_TEST_ATTRIBUTE.replace("{type}", category.name())
+        );
+
+        final List<DataEntityPojo> dataEntityPojos = List.of(new DataEntityPojo()
+                .setOddrn(UUID.randomUUID().toString())
+                .setTypeId(DataEntityTypeDto.JOB.getId())
+                .setSpecificAttributes(jsonb));
+
+        return dataEntityRepository
+                .bulkCreate(dataEntityPojos)
+                .collectList()
+                .block()
+                .get(0);
```

Comment:
I poorly remember how flux works but It feels that those kind of blocking operations are just workarounds to get a single first element from the single element list.

Can we do something like this? Does it work in that way?
pseudo code. I didn't test it.
```suggestion
         return dataEntityRepository
                .create(dataEntityPojo)
                .next()
                .block();
```

Reply:
I mean we can return just `.next()` and treat as a stream further

Reply:
Will be fixed with single object creation

Reply:
done